### ISSUE TITLE
Zipformer_MMI训练问题

### ISSUE THREAD

**Issue by gamesdsad** [k2-fsa/icefall#1962]
您好，我在根据文档训练Zipformer_MMI模型时遇到了如下问题：
当训练从 CTC 损失切换到 MMI 损失后，MMI 损失立即变为 NaN，且多次尝试后问题依然存在，无论将warm_step设置成多少都不可行，且训练过程中最终会出现RuntimeError: grad_scale is too small, exiting: 4.76837158203125e-07的问题。

我的训练版本是torch2.4.0+cu124，CUDA版本是12.4，4*11G_NVIDIA GeForce GTX 1080 Ti   
训练配置如下：
 "attention_dims": "192,192,192,192,192",
  "average_period": 200,
  "base_lr": 0.05,
  "batch_idx_train": 0,
  "best_train_epoch": -1,
  "best_train_loss": Infinity,
  "best_valid_epoch": -1,
  "best_valid_loss": Infinity,
  "bucketing_sampler": true,
  "cnn_module_kernels": "31,31,31,31,31",
  "concatenate_cuts": false,
  "ctc_beam_size": 10,
  "den_scale": 1.0,
  "drop_last": true,
  "duration_factor": 1.0,
  "enable_musan": true,
  "enable_spec_aug": true,
  "encoder_dims": "384,384,384,384,384",
  "encoder_unmasked_dims": "256,256,256,256,256",
  "env_info": {
    "python-version": "3.8",
    "torch-cuda-available": true,
    "torch-cuda-version": "12.4",
    "torch-version": "2.4.0+cu124"
  },
  "exp_dir": "zipformer_mmi/exp",
  "feature_dim": 80,
  "feedforward_dims": "1024,1024,2048,2048,1024",
  "full_libri": false,
  "gap": 1.0,
  "inf_check": false,
  "input_strategy": "PrecomputedFeatures",
  "keep_last_k": 30,
  "lang_dir": "data/lang_bpe_500",
  "log_interval": 50,
  "lr_batches": 5000,
  "lr_epochs": 3.5,
  "manifest_dir": "data/fbank",
  "master_port": 12354,
  "max_duration": 300,
  "mini_libri": false,
  "mmi_beam_size": 6,
  "nhead": "8,8,8,8,8",
  "num_buckets": 30,
  "num_encoder_layers": "2,4,3,2,4",
  "num_epochs": 30,
  "num_workers": 2,
  "on_the_fly_feats": false,
  "print_diagnostics": false,
  "reduction": "sum",
  "reset_interval": 200,
  "return_cuts": true,
  "save_every_n": 2000,
  "seed": 42,
  "shuffle": true,
  "spec_aug_time_warp_factor": 80,
  "start_batch": 0,
  "start_epoch": 1,
  "subsampling_factor": 4,
  "tensorboard": true,
  "use_double_scores": true,
  "use_fp16": true,
  "use_pruned_intersect": false,
  "valid_interval": 1600,
  "vocab_size": 500,
  "warm_step": 2000,
  "world_size": 4,
  "zipformer_downsampling_factors": "1,2,4,8,2"
}

训练日志文件如下：

[train_log.txt](https://github.com/user-attachments/files/20664019/train_log.txt)

我在源代码无修改的情况下，尝试了很多调整训练参数的方法，但都毫无作用，请问该如何解决？
